### PR TITLE
Add figure factory

### PR DESCRIPTION
**Pull request by alena-bartosh**

### CODE REVIEW

File: src/figure.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+from figure_kind import FigureKind
+from point import Point
+
+
+@dataclass
+class Figure:
+    kind: FigureKind
+    position: Point
+    points: list[Point]
```
Comment (s373r) [inline src/figure.py:11]: please extract `list[Point]` as `Points` into `point.py`

File: src/figure_kind_randomizer.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from figure_kind import FigureKind
```
Comment (s373r) [inline src/figure_kind_randomizer.py:1]: please review this file since it is out of scope

File: src/figure_factory.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        FigureKind.SNAKE_RIGHT: [Point(0, 1), Point(0, 2), Point(1, 0), Point(1, 1)],
+        FigureKind.SNAKE_LEFT: [Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 2)],
+        FigureKind.CORNER_RIGHT: [Point(0, 0), Point(1, 0), Point(2, 0), Point(2, 1)],
+        FigureKind.CORNER_LEFT: [Point(0, 1), Point(1, 1), Point(2, 0), Point(2, 1)],
+        FigureKind.T_LETTER: [Point(0, 0), Point(0, 1), Point(0, 2), Point(1, 1)],
+    }
+
+    @staticmethod
+    def create(kind: FigureKind) -> Figure:
+        return Figure(kind, Point(0, 0), FigureFactory.kind_figure_points_mapping[kind])
```
Comment (s373r) [inline src/figure_factory.py]: 1) extract `FigureFactory.kind_figure_points_mapping[kind]` to variable e.g. `figure_points`
2) please pay attention: you need to make a deep copy ...
```py
import copy

copy.deepcopy()
```
... of `FigureFactory.kind_figure_points_mapping[kind]`

Otherwise, you change the mapping

File: src/figure_kind.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+
+class FigureKind(Enum):
+    SQUARE = auto()
+    LINE = auto()
+    SNAKE_RIGHT = auto()
+    SNAKE_LEFT = auto()
+    CORNER_RIGHT = auto()
+    CORNER_LEFT = auto()
+    T_LETTER = auto()
```
Comment (s373r) [inline src/figure_kind.py:11]: thanks for naming -- you did the most difficult part (naming)

**Review by s373r — CHANGES_REQUESTED** [review]

File: src/figure.py (hunk 1 above)
Comment (alena-bartosh) [reply src/figure.py:11]: For what? I think `list[Point]` is clear enough

File: src/figure_factory.py (hunk 3 above)
Comment (alena-bartosh) [reply src/figure_factory.py]: How I change the mapping just using it?

File: src/figure_kind_randomizer.py (hunk 2 above)
Comment (alena-bartosh) [reply src/figure_kind_randomizer.py:1]: This file is ok, I want to merge it and assign to Serg

File: src/figure_factory.py (hunk 3 above)
Comment (alena-bartosh) [reply src/figure_factory.py]:
```
a = FigureFactory.create(FigureKind.SQUARE)
a.points = [Point(1, 0)]
print(FigureFactory.kind_figure_points_mapping[FigureKind.SQUARE])  # [Point(x=0, y=0), Point(x=0, y=1), Point(x=1, y=0), Point(x=1, y=1)]
```

Comment (s373r) [reply src/figure_factory.py]: you had overridden the reference not updating the value

instead of `a.points = [Point(1, 0)]` try `a.points.append(Point(-27, -15))`

File: src/figure_kind_randomizer.py (hunk 2 above)
Comment (s373r) [reply src/figure_kind_randomizer.py:1]: > This file is ok, I want to merge it and assign to Serg

do it in another PR please

File: src/figure.py (hunk 1 above)
Comment (s373r) [reply src/figure.py:11]: we had a lot of `list[Point]` so please change

File: src/figure_factory.py (hunk 3 above)
Comment (alena-bartosh) [reply src/figure_factory.py]: Ok, done

File: src/figure_kind_randomizer.py (hunk 2 above)
Comment (alena-bartosh) [reply src/figure_kind_randomizer.py:1]: Lets remove comments then

File: src/figure.py (hunk 1 above)
Comment (alena-bartosh) [reply src/figure.py:11]: Discussed

File: src/figure_factory.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        FigureKind.SNAKE_RIGHT: [Point(0, 1), Point(0, 2), Point(1, 0), Point(1, 1)],
+        FigureKind.SNAKE_LEFT: [Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 2)],
+        FigureKind.CORNER_RIGHT: [Point(0, 0), Point(1, 0), Point(2, 0), Point(2, 1)],
+        FigureKind.CORNER_LEFT: [Point(0, 1), Point(1, 1), Point(2, 0), Point(2, 1)],
+        FigureKind.T_LETTER: [Point(0, 0), Point(0, 1), Point(0, 2), Point(1, 1)],
+    }
+
+    @staticmethod
+    def create(kind: FigureKind) -> Figure:
+        figure_points = copy.deepcopy(FigureFactory.kind_figure_points_mapping[kind])
```
Comment (s373r) [inline src/figure_factory.py]: add a blank line

**Review by s373r — CHANGES_REQUESTED** [review]

**Review by s373r — DISMISSED** [review]

**Review by s373r — DISMISSED** [review]

**Review by s373r — APPROVED** [review]
cool job!

**Review by Serg-Bartosh — APPROVED** [review]
Well done :grinning: :+1:
